app/controllers/foreman_xen/cache_controller.rb

```diff
@@ -0,0 +1,25 @@
+module ForemanXen
+  class CacheController < ::ApplicationController
+
+    before_action :get_compute_resource
+
+    # POST = foreman_xen/cache/refresh
+    def refresh
+      type                = params[:type]
+      unless @compute_resource.respond_to?("#{type}!")
+        process_error(:error_msg => "Error refreshing cache. Method '#{type}!' not found for compute resource" +
+            @compute_resource.name)
+      end
+
+      respond_to do |format|
+        format.json { render :json => @compute_resource.public_send("#{type}!") }
```

Comment:
I think, this is a bit dangerous. Basically you can execute arbitrary methods on a compute resource model object. A better approach would be to whitelist the methods.